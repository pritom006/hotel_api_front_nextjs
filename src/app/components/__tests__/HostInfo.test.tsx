import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // Provides additional matchers like toBeInTheDocument
import HostSection from '../HostInfo';

describe('HostSection', () => {
  const mockHostInfo = {
    name: 'John Doe',
    languages: [
      { id: 1, name: 'English' },
      { id: 2, name: 'French' },
    ],
    contact: 'john.doe@example.com',
  };

  it('renders the host name correctly', () => {
    render(<HostSection hostInfo={mockHostInfo} />);
    expect(screen.getByText(/Hosted by John Doe/i)).toBeInTheDocument();
  });

  it('renders all languages with correct formatting', () => {
    render(<HostSection hostInfo={mockHostInfo} />);
    expect(screen.getByText(/Languages:/i)).toBeInTheDocument();
    expect(screen.getByText('English,')).toBeInTheDocument();
    expect(screen.getByText('French')).toBeInTheDocument();
  });

  it('renders a contact link with the correct email', () => {
    render(<HostSection hostInfo={mockHostInfo} />);
    const contactLink = screen.getByRole('link', { name: /contact host/i });
    expect(contactLink).toHaveAttribute('href', 'mailto:john.doe@example.com');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<HostSection hostInfo={mockHostInfo} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
