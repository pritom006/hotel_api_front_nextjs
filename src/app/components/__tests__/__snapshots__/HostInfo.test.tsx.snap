// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostSection matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="max-w-7xl ml-20 px-8 font-sans"
  >
    <section
      class="border-t border-b border-gray-200 py-8"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-16"
      >
        <div
          class="flex flex-col"
        >
          <h2
            class="text-2xl font-semibold text-gray-900"
          >
            About the host
          </h2>
          <h3
            class="text-xl font-semibold text-gray-900 mt-20"
          >
            Send a message
          </h3>
        </div>
        <div
          class="flex flex-col space-y-8"
        >
          <div
            class="flex items-center"
          >
            <span
              class="text-2xl text-gray-900"
            >
              Hosted by John Doe
            </span>
          </div>
          <div
            class="space-y-2"
          >
            <div
              class="text-2xl text-gray-900"
            >
              Languages:
            </div>
            <div
              class="flex flex-wrap gap-2"
            >
              <span
                class="text-sm text-gray-600"
              >
                English,
              </span>
              <span
                class="text-sm text-gray-600"
              >
                French
              </span>
            </div>
          </div>
          <a
            class="inline-block bg-gray-100 text-blue-600 px-4 py-3 rounded-lg font-medium hover:bg-gray-200 transition-colors text-center"
            href="mailto:john.doe@example.com"
          >
            Contact host
          </a>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
